### ISSUE TITLE
Docker ROUNDCUBEMAIL_DB_TYPE not properly managed

### ISSUE THREAD

**Issue by madmath03** [roundcube/roundcubemail#6147]
The Docker environment parameter `ROUNDCUBEMAIL_DB_TYPE` does not work as described.

The value of `ROUNDCUBEMAIL_DB_TYPE`  is actually never read to define the content of `db_dsnw`.

For instance, consider the following docker-compose:
```yml
version: '2'

services:
  roundcubedb:
    image: postgres:10-alpine
    container_name: roundcubedb
    environment:
        - POSTGRES_DB=roundcubedb
        - POSTGRES_USER=roundcube
        - POSTGRES_PASSWORD=roundcube

  roundcube:
    image: roundcube/roundcubemail:elastic
    container_name: roundcube
    links:
      - roundcubedb
    environment:
      - ROUNDCUBEMAIL_DB_TYPE=pgsql
      - ROUNDCUBEMAIL_DB_HOST=roundcubedb
      - ROUNDCUBEMAIL_DB_NAME=roundcubedb
      - ROUNDCUBEMAIL_DB_USER=roundcube
      - ROUNDCUBEMAIL_DB_PASSWORD=roundcube
      #...

```

Here is the extract of `config.inc.php` generated on first startup:
```php
<?php
    $config['db_dsnw'] = 'pgsql:////var/www/html/db/roundcubedb.db?mode=0646';
    $config['db_dsnr'] = '';

```

So not what I would expect...

For those who might encounter this issue, note that there is a workaround.
It is possible to trigger the appropriate database connection string generation by adding a `POSTGRES_ENV_POSTGRES_SOMETHING`.
For instance, the following actually generates a valid config:
```yml
version: '2'

services:
  roundcubedb:
    image: postgres:10-alpine
    container_name: roundcubedb
    environment:
        - POSTGRES_DB=roundcubedb
        - POSTGRES_USER=roundcube
        - POSTGRES_PASSWORD=roundcube

  roundcube:
    image: roundcube/roundcubemail:elastic
    container_name: roundcube
    links:
      - roundcubedb
    environment:
      - ROUNDCUBEMAIL_DB_TYPE=pgsql
      - ROUNDCUBEMAIL_DB_HOST=roundcubedb
      - ROUNDCUBEMAIL_DB_NAME=roundcubedb
      - ROUNDCUBEMAIL_DB_USER=roundcube
      - ROUNDCUBEMAIL_DB_PASSWORD=roundcube
      # Here is the trick
      - POSTGRES_ENV_POSTGRES_ENABLED=TRUE
      #...

```

Result:
```php
<?php
    $config['db_dsnw'] = 'pgsql://roundcube:roundcube@roundcubedb/roundcubedb';
    $config['db_dsnr'] = '';

```

The error comes from https://github.com/roundcube/roundcubemail/blob/master/docker/docker-entrypoint.sh#L18
I'm guessing it must _work_ with a link to MySQL container because the roundcube container can somehow see some `MYSQL_ENV_MYSQL_*` variables inside the MySQL container, am I right ?

I would prefer if the `ROUNDCUBEMAIL_DB_TYPE` value was the trigger instead of using some external environment variables. This would allow the container to work with an existing database on a different server and makes much more sense to me.

 @thomascube, @J0WI: your thoughts ?
I do not mind helping to fix this if you guys do not have the time.

**Comment by J0WI** [comment]
> Environment variables are only populated if you’re using the legacy version 1 Compose file format.

https://docs.docker.com/compose/link-env-deprecated/

**Comment by J0WI** [comment]
The `elastic` tag seems not to published here on GitHub btw.

**Comment by madmath03** [comment]
So... you're saying the current mechanism for setting the DB type only works with docker-compose v1 ?
If that's the case, even more reason to change the behavior in `docker-entrypoint.sh` ?

I wanted to see what `elastic` looked like and admit I haven't tested this behavior with non elastic tags.
I will it a try with 1.3.4 and see if I get the same issue.

**Comment by simongreen-net** [comment]
https://github.com/roundcube/roundcubemail/pull/6155

**Comment by thomascube** [comment]
Should be resolved in PR #6155 (128587d0d)
